File: CheckMyStar.Bll.Models/UserModel.cs
using CheckMyStar.Enumerations;

namespace CheckMyStar.Bll.Models
{
    public class UserModel
    {
        public int Identifier { get; set; }
        public EnumCivility Civility { get; set; }
        public string? LastName { get; set; }
        public string? FirstName { get; set; }
        public int? SocietyIdentifier { get; set; }
        public string? Email { get; set; }
        public string? Phone { get; set; }
        public string? Password { get; set; }
        public EnumRole Role { get; set; }
        public AddressModel? Address { get; set; }
        public bool IsActive { get; set; }
        public bool IsFirstConnection { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime UpdatedDate { get; set; }
    }
}

File: CheckMyStar.Angular/src/app/www/20_Models/Common/User.model.ts
import { EnumCivility } from '../../10_Common/Enumerations/EnumCivility';
import { EnumRole } from '../../10_Common/Enumerations/EnumRole';
import { AddressModel } from './Address.model';


export interface UserModel {
    identifier: number;
    civility: EnumCivility;
    lastName: string;
    firstName: string;
    societyIdentifier?: number | null;
    email: string;
    phone: string;
    password: string;
    role: EnumRole;
    address: AddressModel;
    isActive: boolean;
} 